lib/benchee/conversion.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Benchee.Conversion do
+  @moduledoc """
+  Integration of the conversion of multiple units with benchee.
+
+  Can be used by plugins to use benche unit scaling logic.
+  """
+
+  alias Benchee.Conversion.{Duration, Count}
+
+  @doc """
+  Takes scenarios and a given scaling_strategy, returns the best unit for
+  run_time and ips, according to the scaling_strategy, in a map.
+
+  The units can then be passed on to the appropriate `format` calls to format
+  the output of arbitrary values with the right unit.
+
+  ## Examples
+
+      iex> statistics = %Benchee.Statistics{average: 1000.0, ips: 1000.0}
+      iex> scenario = %Benchee.Benchmark.Scenario{run_time_statistics: statistics}
+      iex> Benchee.Conversion.run_time_units([scenario], :best)
+      %{
+        ips:      %Benchee.Conversion.Unit{
+                    label: "K",
+                    long: "Thousand",
+                    magnitude: 1000,
+                    name: :thousand
+                  },
+        run_time: %Benchee.Conversion.Unit{
+                    label: "ms",
+                    long: "Milliseconds",
+                    magnitude: 1000,
+                    name: :millisecond
+                    }
+      }
+  """
+  def run_time_units(scenarios, scaling_strategy) do
```

Comment:
This name is off because `run_time` is one of the units returned, but there are other units too. I’d call it either `units` and pattern match on the list of scenarios to determine what units are appropriate, or `scenario_units`

Reply:
true that, must have been late yesterday. My thinking was: this only deals with `run_time_statistics` and soon we'll also have memory_statistics. Of course, this would still return both so will rename back to the original `units` - thanks! :)